runtime/protocol/http-client-engines/http-client-engine-crt/jvm/src/aws/smithy/kotlin/runtime/http/engine/crt/ConnectionManager.kt

```diff
@@ -82,8 +93,18 @@ internal class ConnectionManager(
             }
 
             throw httpEx
+        } finally {
+            metrics.queuedRequests = pending.decrementAndGet()
+            emitMetrics()
         }
     }
+
+    private fun emitMetrics() {
+        val acquiredConnections = connManagers.values.sumOf { it.managerMetrics.leasedConcurrency }
+        metrics.acquiredConnections = acquiredConnections
+        metrics.idleConnections = config.maxConnections.toLong() - acquiredConnections
```

Comment:
**fix**: This still isn't right, idle connections are connections established with the server not currently in use. If I'm not mistaken this is just [availableConcurrency](https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/http/HttpManagerMetrics.java#L22C2-L22C2) which would require the same summing across managers.

Reply:
OK, sum of `availableConcurrency` sounds doable, although it will exceed max connections under some circumstances.